### PR TITLE
[#4022] Throwing Generic Exceptions - Adaptive Expressions

### PR DESCRIPTION
**Pull request by santgr11**
Addresses # 4022

## Description

This PR replaces all the generic expressions in `AdaptiveExpressions`, `Adapters`, `integration`, `FunctionalTests`, and part of `BotBuilder` libraries. Also, updates the affected testes to assert the proper exception.

## Specific Changes
Replaced generic exceptions in:
- FunctionalTests
- libraries\Adapters
- libraries\AdaptiveExpressions
- libraries\integration
- libraries\Microsoft.Bot.Builder\Adapters

Updated tests to assert proper exception:
- tests\Microsoft.Bot.Builder.Tests\Adapters\TestFlowTests.cs

## Testing
image

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs [hunk 1]
```diff
@@ -280,14 +280,14 @@ public TestFlow AssertReply(IActivity expected, IEqualityComparer<IActivity> equ
...
-                        throw new Exception($"{description}: Type should match");
+                        throw new ArgumentException($"{description}: Type should match");
                     }
 
                     if (equalityComparer != null)
                     {
                         if (!equalityComparer.Equals(expected, reply))
                         {
-                            throw new Exception($"Expected:{expected}\nReceived:{reply}");
+                            throw new ArgumentException($"Expected:{expected}\nReceived:{reply}");
```
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs:290]: I'm not sure these should be ArgumentExceptions. Maybe `InvalidOperationException` suits better for assertions.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs (hunk 1 above)
Comment (iscaltritti-sw) [reply libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs:290]: I think `ArgumentException` is more specific, but both work fine.

**Review by iscaltritti-sw — APPROVED** [review]

File: libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs (hunk 1 above)
Comment (santgr11) [reply libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs]: fixed, thanks

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Comment by ceciliaavila** [comment]
Promoted to MS in # 4951

**Pull request closed without merging**